labelbox/data/serialization/ndjson/label.py

```diff
@@ -19,23 +20,58 @@
 from .metric import NDScalarMetric, NDMetricAnnotation, NDConfusionMatrixMetric
 from .classification import NDChecklistSubclass, NDClassification, NDClassificationType, NDRadioSubclass
 from .objects import NDObject, NDObjectType, NDSegments, NDDicomSegments, NDVideoMasks, NDDicomMasks
+from .relationship import NDRelationship
 from .base import DataRow
 
 
 class NDLabel(BaseModel):
     annotations: List[Union[NDObjectType, NDClassificationType,
                             NDConfusionMatrixMetric, NDScalarMetric,
                             NDDicomSegments, NDSegments, NDDicomMasks,
-                            NDVideoMasks]]
+                            NDVideoMasks, NDRelationship]]
+
+    class _RelationshipTuple(BaseModel):
+        ndjson: NDRelationship
+        source: str
+        target: str
+
+    class _AnnotationGroupTuple(BaseModel):
+        data_row: DataRow = None
+        ndjson_objects: Dict[str,
+                             Union[NDObjectType, NDClassificationType,
+                                   NDConfusionMatrixMetric, NDScalarMetric,
+                                   NDDicomSegments, NDSegments, NDDicomMasks,
+                                   NDVideoMasks, NDRelationship]] = {}
+        relationships: List["NDLabel._RelationshipTuple"] = []
+        annotations: Dict[str, ObjectAnnotation] = {}
 
     def to_common(self) -> LabelGenerator:
-        grouped_annotations = defaultdict(list)
-        for annotation in self.annotations:
-            grouped_annotations[annotation.data_row.id or
-                                annotation.data_row.global_key].append(
-                                    annotation)
+        annotation_groups = defaultdict(lambda: NDLabel._AnnotationGroupTuple())
```

Comment:
This code was pretty difficult for me to follow. I think it could benefit from more comments or even a detailed docstring explaining the logic behind constructing `_AnnotationGroupTuple`

Reply:
That makes sense, I will add comments/docstring to make this clearer.

Reply:
good catch! can't remember why I added it and then I forgot about it, in this case, it's not necessary and I'll remove it.